### PR TITLE
Add Typos to CI

### PR DESCRIPTION
**Pull request by Hug0-Drelon**
## What?
Fixes partially [#2003](https://github.com/polylang/polylang-pro/issues/2003)

### CODE REVIEW

File: tests/phpunit/tests/test-wpml.php [hunk 1]
```diff
@@ -514,7 +514,7 @@ public function test_wpml_get_element_translations_for_terms() {
 		$en = self::factory()->term->create( array( 'name' => 'Sample tag' ) );
 		self::$model->term->set_language( $en, 'en' );
 
-		$fr = self::factory()->term->create( array( 'name' => 'Etiquette exemple' ) );
+		$fr = self::factory()->term->create( array( 'name' => 'Etiquette example' ) );
```
Comment (Screenfeed) [inline tests/phpunit/tests/test-wpml.php:517]: Sad 😢

**Review by Screenfeed — CHANGES_REQUESTED** [review]

File: .typos.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "olt-manager\\.php",
+]
+
+[default.extend-words]
+
+[default.extend-identifiers]
+"OLT_Manager_Test" = "OLT_Manager_Test"
+"PLL_OLT" = "PLL_OLT"
+"PLL_OLT_Manager" = "PLL_OLT_Manager"
+"get_post_stati" = "get_post_stati"
```
[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply .typos.toml]: Weird. Changed in d53ee686.

File: .typos.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[files]
+extend-exclude = [
+    ".git/",
+    "flags/??.png",
+    "settings/flags.php",
+    "settings/languages.php",
+	"vendor/",
+	"node_modules/",
```
[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply .typos.toml]: Done in d53ee686, except for `.typo.toml `.

File: tests/phpunit/tests/test-wpml.php (hunk 1 above)
Comment (Hug0-Drelon) [reply tests/phpunit/tests/test-wpml.php]: Set back and ignored in d53ee686.

Comment (Screenfeed) [reply tests/phpunit/tests/test-wpml.php]: Oh, you found a way, nice.

File: .typos.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply .typos.toml]: Why not?

[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply .typos.toml]: Typos doesn't parse its config file.

[the remark this answers is not in the document]
Comment (Screenfeed) [reply .typos.toml]: The VSCode plugin does x)

[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply .typos.toml]: OK added in [a1c4540](https://github.com/polylang/polylang/pull/1433/commits/a1c45409b374f4d09c6c3c7ae7b43c2be4badf31).

[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply .typos.toml]: Even though it can be excluded directly from the extension.

[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply .typos.toml]: In the meantime, it's better to have it set in one place for several devs than on each of their machines.

**Review by Chouby — APPROVED** [review]

File: .typos.toml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+extend-exclude = [
+    ".git/",
+    ".phpunit.result.cache",
+    "flags/??.png",
+    "node_modules/",
+    "settings/flags.php",
+    "settings/languages.php",
+    "tmp/",
+    "vendor/",
+	".typos.toml",
```
Comment (szepeviktor) [inline .typos.toml:11]: The config file is always excluded.
```suggestion
```

Comment (Hug0-Drelon) [reply .typos.toml:11]: That's true, but the Typos VS Code extension doesn't. Cf this [discussion](https://github.com/polylang/polylang/pull/1433#discussion_r1514721022).

Comment (szepeviktor) [reply .typos.toml:11]: Reported in https://github.com/tekumara/typos-lsp/issues/47

Comment (Hug0-Drelon) [reply .typos.toml:11]: Thanks @szepeviktor!

Comment (szepeviktor) [reply .typos.toml:11]: @Hug0-Drelon Released in https://github.com/tekumara/typos-lsp/releases/tag/v0.1.16

**Pull request merged**